src/Symfony/Component/Security/Core/Authorization/Voter/Context/VoterContext.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter\Context;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Report\VoteReportBuilder;
+use Symfony\Component\Security\Core\Authorization\Voter\Report\VoteReportCollectorInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * A context created by {@link VoterContextFactory} and
+ * used for {@link Voter}.
+ *
+ * @author Maxime Perrimond <max.perrimond@gmail.com>
+ */
+class VoterContext implements VoterContextInterface
+{
+    /**
+     * @var string
+     */
+    protected $currentAttribute;
+
+    /**
+     * @var mixed
+     */
+    protected $subject;
+
+    /**
+     * @var TokenInterface
+     */
+    protected $token;
+
+    /**
+     * @var VoterInterface
+     */
+    protected $voter;
+
+    /**
+     * @var VoteReportCollectorInterface
+     */
+    protected $collector;
+
+    /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
+
+    /**
+     * @var null|string
+     */
+    protected $translationDomain;
```

Comment:
Unless there's a very good reason to make them protected, please make properties private